core/src/main/kotlin/io/rover/campaigns/core/CoreAssembler.kt

```diff
@@ -479,7 +483,7 @@ class CoreAssembler @JvmOverloads constructor(
             WorkManager.getInstance().cancelAllWorkByTag("rover-sync")
         }
 
-        application.registerActivityLifecycleCallbacks(AutoScreenTracker())
+        application.registerActivityLifecycleCallbacks(AutoScreenTracker(analyticsActivityAutoTracking))
```

Comment:
ah, rather than passing it through, and still creating the lifecycle callback listener and all that jazz, why not just skip `application.registerActivityLifecycleCallbacks(AutoScreenTracker())` entirely if `analyticsActivityAutoTracking` is false?

Reply:
😄 🤦‍♂